servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRoutes.java

```diff
@@ -173,9 +187,18 @@ protected GrpcService newServiceFromRoutes(final AllGrpcRoutes routes) {
     }
 
     @Nullable
-    private GrpcExecutionStrategy executionStrategy(final Method method, final Class<?> clazz) {
-        return getAndValidateRouteExecutionStrategyAnnotationIfPresent(method, clazz, strategyFactory, errors,
-                noOffloadsStrategy());
+    private GrpcExecutionStrategy executionStrategy(final String path, final Method method, final Class<?> clazz) {
+        // Check if we already have a computed GrpcExecutionStrategy for this path. This may happen when we re-register
+        // filtered routes:
+        final GrpcExecutionStrategy saved = executionStrategies.getOrDefault(path, NULL);
+        if (saved != NULL) {
+            return saved;
+        }
+        final GrpcExecutionStrategy computed = getAndValidateRouteExecutionStrategyAnnotationIfPresent(method, clazz,
+                strategyFactory, errors, noOffloadsStrategy());
+        // Save computed value even if it's null to prevent analysing annotations on a service filter class
+        executionStrategies.put(path, computed);
```

Comment:
@NiteshKant should we account for users that may try-catch routes registration? If they register the same path 2+ times with different route implementations, we will override the strategy here, but later we throw an exception saying that this path was already registered. If users try-catch this exception, internal state will be modified. I can move this `executionStrategies.put` after we register a route if necessary.

Reply:
>> but later we throw an exception saying that this path was already registered.

Did you mean that we will ignore the strategy for a subsequent registration? We are not using `executionStrategies` to check for duplicate registration, are we?

I think we should account for cases when a registration exception is caught and retried unless we have a strong reason to believe that is invalid. AFAIR, this is required only for disallowing filters to specify the strategy annotation, rite? I have not thought of that case as to whether it is good or bad, so if we need special code to disallow it, lets refrain from doing that for now and have "behavior when strategy annotation is defined on filters is undefined" as our answer.

Reply:
I was thinking about a use-case like this:
```java
ServiceFactory.Builder builder = new ServiceFactory.Builder();
builder.testBlocking(noOffloadsStrategy(), serviceImpl);
try {
    builder.testBlocking(anotherStrategy, anotherServiceImpl);
} catch (IllegalStateException e) {
    // ignore that this route was already registered
}
```
Don't think we need to verify duplication for the `executionStrategies` map, because we already do that for routes. An additional check will be useless.
In my first impl I stored execution strategy before I verified routes duplication. That's why in the example above users will be in the state when they have `serviceImpl` with `anotherStrategy`.
Now I moved `executionStrategies.put(...)` after we verify routes duplication. So, nothing will corrupt the internal state.

> AFAIR, this is required only for disallowing filters to specify the strategy annotation, rite? I have not thought of that case as to whether it is good or bad.

Right. As we discussed before, ST policy is that if you configure an execution strategy for the route, ST will use it regardless of filters/influencers/etc.
The case here is that grpc filter implements the same service interface. Therefore, users can accidentally add `@RouteExecutionStrategy` annotation on the filter. I think we should ignore it.

> so if we need special code to disallow it, lets refrain from doing that for now and have "behavior when strategy annotation is defined on filters is undefined" as our answer.

The code to disallow annotation inspection for filters is trivial. It's just `getOrDefault` vs `get` on the map to distinguish `key-null` entry vs entry for this key does not exist.

Defined behavior is always better than undefined. I've added this test to demonstrate the problem: https://github.com/apple/servicetalk/pull/932/files#diff-4fbfd0eed0161987f94f03bc6d9e3203R262-R279